doc/patch-author-guide.md

```diff
@@ -214,6 +214,15 @@ safe to access:
  	 * Flush inherited counters to the parent - before the parent
  	 * gets woken up by child-exit notifications.
 ```
+Notes:
+* `kpatch_shadow_alloc` initializes only shadow variable metadata. It
+  allocates variable storage via `kmalloc` with the `gfp_t` flags it is
+  given, but otherwise leaves the area untouched. Initialization of a shadow
+  variable is the responsibility of the caller.
+* As soon as `kpatch_shadow_alloc` creates a shadow variable, its presence
+  will be reported by `kpatch_shadow_get`. Care should be taken to avoid any
+  potential race conditions between a kernel thread that allocates a shadow
+  variable and concurrent threads that may attempt to use it.
```

Comment:
Here it might be good to describe our solution of splitting it up into two patches.  (The second patch could still be a superset of the first so it would still be compatible with our recommendation of cumulative patches.)